File: TwoPointers/js/isSubsequence.js
const isSubsequence=(string1, string2)=>{
    let i = 0
    let j = 0

    while(i<string1.length && j < string2.length){
        if(string1[i] === string2[j]){
            i +=1
            j +=1
        }else{
            j +=1
        }
    }

    return i === string1.length
} 

File: TwoPointers/go/is_subsequence.go
package twopointers

// IsSubsequence reports whether s1 appears in s2 in order.
func IsSubsequence(s1, s2 string) bool {
	i, j := 0, 0
	for i < len(s1) && j < len(s2) {
		if s1[i] == s2[j] {
			i++
			j++
		} else {
			j++
		}
	}
	return i == len(s1)
}
